docs/current/guides/541047-alternative-runtimes.md

```diff
@@ -0,0 +1,52 @@
+---
+slug: /541047/alternative-runtimes
+displayed_sidebar: 'current'
+category: "guides"
+tags: ["podman"]
+authors: ["Vikram Vaswani"]
+date: "2023-04-28"
+---
+
+# Use Dagger with Alternative OCI Runtimes
+
+## Introduction
+
+This guide explains how to use Dagger with various OCI-compatible Docker alternatives.
+
+## Podman
+
+### Requirements
+
+This guide assumes that you have Podman installed and running on the host system. If not, [install Podman](https://podman.io/getting-started/installation).
+
+### Configuration
+
+By default, Dagger tries to invoke the `docker` executable. To use Podman instead, create a symbolic link to the Podman executable in your system path and name it `docker`:
+
+```shell
+sudo ln -s `which podman` /usr/bin/docker
```

Comment:
`$()` 👉 https://github.com/koalaman/shellcheck/wiki/SC2006#rationale

What happens if `$(which podman)` fails? What happens if `/usr/bin/docker` does not exist? FWIW:

```console
> sudo ln -s `which podman` /usr/bin/docker
Password:
ln: target '/usr/bin/docker': No such file or directory
```

[`/usr/local/bin`](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html) is usually used for this purpose.

I would be tempted to leave here just an example. Something along the lines of:

```console
# Check that podman is installed and available in your shell PATH:
podman version

# E.g.
# Client:       Podman Engine
# Version:      4.5.0
# API Version:  4.5.0
# Go Version:   go1.20.3
# Git Commit:   75e3c12579d391b81d871fd1cded6cf0d043550a
# Built:        Fri Apr 14 14:28:20 2023
# OS/Arch:      darwin/amd64
# 
# Server:       Podman Engine
# Version:      4.5.0
# API Version:  4.5.0
# Go Version:   go1.20.2
# Built:        Fri Apr 14 16:42:22 2023
# OS/Arch:      linux/amd64

# Alias docker to podman
alias docker=podman

# Check that docker version command works the same as podman version
docker version

# If everything works as expected in your current shell session, consider making it a permanent configuration by adding `alias docker=podman` to your shell configuration, e.g. `$HOME/.zshrc`. `$HOME/.bashrc`, etc.
```

FWIW, `alias` has a higher precedence than a PATH executable. Using `alias` avoids a bunch of issues like `/usr/local/bin/docker` already existing, PATH ordering, etc.

Reply:
We (@jpadams  and I, separately) tried `alias` already but it doesn't work. Details in https://discord.com/channels/707636530424053791/1100004042207416404/1100290383067488318

Please let me know how to proceed?